ocs_ci/ocs/ui/acm_ui.py

```diff
@@ -103,3 +130,487 @@ def navigate_credentials_page(self):
+        self.cluster_path = self.cluster_conf.ENV_DATA["cluster_path"]
+        self.deploy_sync_mode = config.MULTICLUSTER.get("deploy_sync_mode", "async")
+        self.deployment_status = None
+        self.cluster_deploy_timeout = self.cluster_conf.ENV_DATA.get(
+            "cluster_deploy_timeout", ACM_CLUSTER_DEPLOY_TIMEOUT
+        )
+        self.deployment_failed_reason = None
+        self.deployment_start_time = 0
+
+    def create_cluster_prereq(self):
+        raise NotImplementedError("Child class has to implement this method")
+
+    def navigate_create_clusters_page(self):
+        # Navigate to Clusters page which has 'Create Cluster'/
+        # 'Import Cluster' buttons
+        # Here we click on "Create Cluster" and we will be in create cluster page
+        while True:
+            self.navigate_clusters_page()
+            log.info("Clicking on 'CreateCluster'")
+            if not self.check_element_presence(
+                (By.XPATH, self.acm_page_nav["cc_create_cluster"][0]), timeout=200
+            ):
+                raise ACMClusterDeployException("Create cluster button not found")
+            log.info("Found create cluster")
+            if not self.check_element_presence(
+                (By.XPATH, self.acm_page_nav["cc_create_cluster_index_xpath"][0]),
+                timeout=300,
+            ):
+                raise ACMClusterDeployException("Create cluster button not found")
+            log.info("Found create cluster by index path")
+            self.do_click(locator=self.acm_page_nav["cc_create_cluster"], timeout=100)
+            time.sleep(20)
+            if self.driver.current_url.endswith("create-cluster"):
+                break
+
+    def click_next_button(self):
+        self.do_click(self.acm_page_nav["cc_next_page_button"])
+
+    def send_keys_multiple(self, key_val):
+        """
+        In a page if we want to fill multiple text boxes we can use
+        this function which iteratively fills in values from the dictionary parameter
+
+        key_val (dict): keys corresponds to the xpath of text box, value corresponds
+            to the value to be filled in
+
+        """
+        for xpath, value in key_val.items():
+            self.do_send_keys(locator=xpath, text=value)
+
+    def click_platform_and_credentials(self):
+        self.navigate_create_clusters_page()
+        self.do_click(
+            locator=self.acm_page_nav[PLATFORM_XPATH_MAP[self.platform]], timeout=100
+        )
+        self.do_click(
+            locator=self.acm_page_nav["cc_infrastructure_provider_creds_dropdown"]
+        )
+        credential = format_locator(
+            self.acm_page_nav["cc_infrastructure_provider_creds_select_creds"],
+            self.platform_credential_name,
+        )
+        self.do_click(locator=credential)
+
+    @retry(ACMClusterDeployException, tries=3, delay=10, backoff=1)
+    def goto_cluster_details_page(self):
```

Comment:
At some places even its single line I want to convert it into function because I can then use retry decorator on that function as you know some of the problems were unsolved even using wait functions from base UI.

Reply:
Ack